### PR TITLE
Querier/Ruler: add histogram to track fetched chunk size distribution

### PR DESCRIPTION
**Pull request by dannykopping**
**What this PR does / why we need it**:
We are looking at making some changes to the chunk caching strategy, and we need this data to know what size chunks we're typically requesting. For example, if the overwhelming majority of chunks are small we may decide to cache only small chunks to decrease the number of requests to the object store; we could fit more chunks in cache this way, which may have a positive performance impact.

**Which issue(s) this PR fixes**:
N/A

**Special notes for your reviewer**:
I picked buckets that I thought would be useful, and even included a bucket that is above the recommended chunk target size since some users may be setting it that high.

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [ ] Documentation added
- [ ] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: pkg/storage/chunk/fetcher/fetcher.go [hunk 1]
```diff
@@ -173,6 +183,11 @@ func (c *Fetcher) FetchChunks(ctx context.Context, chunks []chunk.Chunk, keys []
 	if err != nil {
 		level.Warn(log).Log("msg", "error fetching from cache", "err", err)
 	}
+
+	for _, buf := range cacheBufs {
+		chunkFetchedSize.With(labels.FromStrings("source", "cache").Map()).Observe(float64(len(buf)))
```
Comment (cyriltovena) [inline pkg/storage/chunk/fetcher/fetcher.go:188]: `WithValues` should be more performant than creating a new labels set and turning it into a map.

**Review by cyriltovena — APPROVED** [review]
LGTM

Please consider my comment.

**Review by sandeepsukhani — APPROVED** [review]
left a minor non-blocking nit, but otherwise it LGTM

File: pkg/storage/chunk/fetcher/fetcher.go (hunk 1 above)
Comment (sandeepsukhani) [reply pkg/storage/chunk/fetcher/fetcher.go:188]: hah, you beat me

**Review by vlad-diachenko — APPROVED** [review]
LGTM

File: pkg/storage/chunk/fetcher/fetcher.go (hunk 1 above)
Comment (dannykopping) [reply pkg/storage/chunk/fetcher/fetcher.go]: Doh! Thanks for spotting this

**Pull request merged**